fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/predicates/RangeConstraints.java

```diff
@@ -353,6 +353,29 @@ public RangeConstraints rebase(@Nonnull final AliasMap translationMap) {
     @Nonnull
     public RangeConstraints translateCorrelations(@Nonnull final TranslationMap translationMap,
                                                   final boolean shouldSimplifyValues) {
+        //
```

Comment:
Just curious, how did you stumble over this problem?

Reply:
Some of the rewrites that happen as a part of translating the values boxes hit this. For example, if you had something like:

```sql
SELECT .... FROM value(x = 42) AS v, T
   WHERE T.a = v.x
```

Then when this gets translated into:

```sql
SELECT .... FROM T
   WHERE T.a = 42
```

The range comparison on `T.a` goes from being a deferred range to a compilable comparison. Some of the test cases I wrote hit that exact problem when it was trying to make its asserts